src/mbgl/style/layers/custom_layer_impl.cpp

```diff
@@ -43,6 +43,11 @@ void CustomLayer::Impl::initialize() {
     initializeFn(context);
 }
 
+void CustomLayer::Impl::deinitialize() {
+    assert(deinitializeFn);
+    deinitializeFn(context);
```

Comment:
Also need to remove deinitialization from `CustomLayer::Impl::~Impl`. And from the looks of that method, it's possible that `deinitializeFn` may be optional? I don't remember but we should probably keep the guard.